src/client/auth/oidc.rs

```diff
@@ -177,13 +159,81 @@ pub struct IdpServerResponse {
     pub refresh_token: Option<String>,
 }
 
+pub(crate) fn build_speculative_client_first(credential: &Credential) -> Command {
+    self::build_client_first(credential, None)
+}
+
+/// Constructs the first client message in the OIDC handshake for speculative authentication
+pub(crate) fn build_client_first(
+    credential: &Credential,
+    server_api: Option<&ServerApi>,
+) -> Command {
+    let mut auth_command_doc = doc! {
+        "authenticate": 1,
+        "mechanism": MONGODB_OIDC_STR,
+    };
+
+    if credential.oidc_callback.is_none() {
+        auth_command_doc.insert("jwt", "");
+    } else if let Some(access_token) = credential
```

Comment:
```suggestion
    } else if let Some(access_token) = credential.get_access_token()
```

(looks like this chain of method calls is the same as the contents of `get_access_token`)

Reply:
Yep! Great call!